packages/cli/src/common/zip.js

```diff
@@ -0,0 +1,43 @@
+const path = require("path");
+const fs = require("fs");
+const os = require("os");
+const archiver = require("archiver");
+
+/**
+ * Compress the directory contents of `directoryPath` and creates a zip archive in the os temp dir
+ * @param {string} directoryPath - Directory to compress
+ * @returns {Promise<string>} - Promise with the path of zip file created
+ */
+function zipDirectory(directoryPath) {
+  const tempDir = os.tmpdir();
+  if (!fs.existsSync(tempDir)) {
```

Comment:
Using `exists` or `existsSync` in this way isn't recommended.  From the [node fs]() docs:

> Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.

With that info, I suggest removing the if statement and calling `fs.mkdirSync` within a try/catch instead, and handling any non-existence errors in the catch.